cmd/query/app/static_handler.go

```diff
@@ -166,35 +173,35 @@ func (sH *StaticAssetsHandler) configListener(watcher *fsnotify.Watcher) {
 	}
 }
 
-func (sH *StaticAssetsHandler) watch() {
+func (sH *StaticAssetsHandler) watch() error {
 	if sH.options.UIConfigPath == "" {
-		return
+		sH.options.Logger.Info("UI config path not provided, config file will not be watched")
+		return nil
 	}
 
-	watcher, err := fsnotify.NewWatcher()
+	watcher, err := sH.watcherFactory.NewWatcher()
 	if err != nil {
-		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
-		return
+		sH.options.Logger.Error("failed to create a new Watcher for the UI config", zap.Error(err))
+		return err
 	}
 
 	go func() {
 		sH.configListener(watcher)
 	}()
 
-	err = watcher.Add(sH.options.UIConfigPath)
-	if err != nil {
-		sH.options.Logger.Error("error adding watcher to file", zap.String("file", sH.options.UIConfigPath), zap.Error(err))
-	} else {
-		sH.options.Logger.Info("watching", zap.String("file", sH.options.UIConfigPath))
+	if err := watcher.Add(sH.options.UIConfigPath); err != nil {
+		sH.options.Logger.Error("error adding Watcher to file", zap.String("file", sH.options.UIConfigPath), zap.Error(err))
+		return err
```

Comment:
nit: both logging and returning an error is anti-pattern. I would wrap and return it, and only log once in the ctor.